### PR TITLE
Add swapfile service

### PR DESCRIPTION
**Pull request by rickynils**
This is a oneshot service that tries to activate a swapfile.
If the file doesn't exist it is created first. Its size is set to
min(disksize(swapfile path) / 2, memsize)

### CODE REVIEW

**Comment by fmap** [comment]
I think `systemd` can handle most of the logic here for you; imagine two separate units, the first a `oneshot` building the file (having `ConditionPathExists= "!${cfg.filePath}"`), the second a dependent [swap unit](http://www.freedesktop.org/software/systemd/man/systemd.swap.html) with `What` set to its artifact.

Then again, there are always more yaks.

**Comment by rickynils** [comment]
@fmap You are so right. Let me at least groom that yak a little bit.

**Comment by proger** [comment]
If `systemd` is Turing-complete we should start writing things with it :P

**Comment by rickynils** [comment]
OK, I'm not going to bother with swap units until NixOS has support for them. Otherwise I have to generate the unit files myself and then this simple thing gets a bit convoluted.

**Comment by fmap** [comment]
:ok_hand:

File: nixos/ec2/autoswap.nix [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+let
+
+  cfg = config.ec2.setupSwapFile;
+
+  inherit (lib) mkOption types;
+
+in {
+
+  options = {
+    ec2.setupSwapFile = {
```
Comment (proger) [inline nixos/ec2/autoswap.nix]: does it have to be `ec2`?

Comment (rickynils) [inline nixos/ec2/autoswap.nix]: No, I'll change it. I think named like this because I hard-coded `/disk0` before.

**Pull request merged**